### PR TITLE
fix bug: previously clicking cancel on admin details page opens edit panel

### PR DESCRIPTION
**Pull request by annyhe**
- fix UI bug
- add tests to show bug is fixed
- introduce airbnb enzyme to react unit testing. 

Why enzyme?
- more concise, easier to read tests --> easier to maintain
- .find(className or nodeName) to find DOM element in rendered tree,
vs. TestUtils verbose props.children...props.children
- based off of React.TestUtils 
- relatively stable (v.2.6) and well-documented:
http://brewhouse.io/2016/03/18/accelerate-your-react-testing-with-enzyme.html
https://medium.com/airbnb-engineering/enzyme-javascript-testing-utilities-for-react-a417e5e5090f#.xyjupa296

### CODE REVIEW

**Comment by annyhe** [comment]
Review app at http://refocus-pr-91.herokuapp.com//subjects/Salesforce.SFDC_Core.WAX.SPX.NA21

File: package.json [hunk 1]
```diff
@@ -65,6 +65,7 @@
     "connect-redis": "^3.0.2",
     "cookie-parser": "^1.4.1",
     "debug": "^2.2.0",
+    "enzyme": "^2.6.0",
```
Comment (iamigo) [inline package.json]: @meynet license ok for this new external dependency?

**Comment by pallavi2209** [comment]
@annyhe When I go to http://refocus-pr-91.herokuapp.com/subjects/Salesforce.SFDC_Core.WAX.SPX.CS33 and click Add child, then cancel, I am redirected to subject list page. Is that expected?

**Comment by iamigo** [comment]
> When I go to http://refocus-pr-91.herokuapp.com/subjects/Salesforce.SFDC_Core.WAX.SPX.CS33 and click Add child, then cancel, I am redirected to subject list page. Is that expected?

I'm ok with having it like it is for right now--it's a UI design question we should let Jee decide when she gets to it.

**Comment by annyhe** [comment]
@pallavi2209 Yes that is the expected behavior, and identical to the UI on the master.

File: package.json (hunk 1 above)
Comment (annyhe) [inline package.json]: @iamigo @meynet Enzyme uses MIT license. https://www.npmjs.com/package/enzyme

**Pull request merged**